app/src/main/java/com/example/itacademyhw/adapter/BeerAdapter.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.itacademyhw.adapter
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.paging.PagingDataAdapter
+import androidx.recyclerview.widget.DiffUtil
+import com.example.itacademyhw.ViewHolders.BeerViewHolder
+import com.example.itacademyhw.databinding.BeerItemBinding
+import com.example.itacademyhw.model.BeerData
+
+class BeerAdapter(
+    context: Context
+): PagingDataAdapter<BeerData, BeerViewHolder>(DIFF_CALLBACK) {
+
+    private val layoutInflater = LayoutInflater.from(context)
+    private val itemClickedCallback: () -> Unit = {
+        //success callback
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BeerViewHolder {
+        return BeerViewHolder(
+                binding = BeerItemBinding.inflate(
+                    layoutInflater,
+                    parent,
+                    false
+                ),
+            callback = itemClickedCallback
+            )
+    }
+
+    override fun onBindViewHolder(holder: BeerViewHolder, position: Int) {
+        val item = getItem(position)
+        when (holder) {
+            is BeerViewHolder -> {
+                holder.bind(item ?: BeerData(0,"","","","", emptyList(),""))
```

Comment:
можно было в классе BeerData определить дефолтные поля и чтобы было лучше можно так
```suggestion
                item?.let(holder::bind)
```
плюс странное поведение будет если пустой элемент покажется где-то в списке, nullable поля нужны в этом адептере чтобы показывать какие-то индикаторы загрузки на элементе (например прогресс или skeleton loading - анимация переливания компонентов с холдере)